pkg/server/apihandlers/message_handlers.go

```diff
@@ -0,0 +1,176 @@
+// @Description get message by session id and message id
+// @Tags messages
+// @Accept json
+// @Produce json
+// @Param sessionId path string true "Session ID"
+// @Param messageId path string true "Message ID"
+// @Success 200 {object} Message
+// @Failure 404 {object} APIError "Not Found"
+// @Failure 500 {object} APIError "Internal Server Error"
+// @Router /api/v1/session/{sessionId}/message/{messageId} [get]
+func GetMessageHandler(appState *models.AppState) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		sessionID := chi.URLParam(r, "sessionId")
+		messageUUID := handlertools.UUIDFromURL(r, w, "messageId")
+		log.Debugf("GetMessageHandler: sessionID: %s, messageID: %s", sessionID, messageUUID)
+
+		messageIDs := []uuid.UUID{messageUUID}
+		messages, err := appState.MemoryStore.GetMessagesByUUID(r.Context(), sessionID, messageIDs)
+		if err != nil {
+			if errors.Is(err, models.ErrNotFound) {
+				handlertools.RenderError(w, fmt.Errorf("not found"), http.StatusNotFound)
+				return
+			} else {
+				handlertools.RenderError(w, err, http.StatusInternalServerError)
+				return
+			}
+		}
+
+		if err := handlertools.EncodeJSON(w, messages[0]); err != nil {
+			handlertools.RenderError(w, err, http.StatusInternalServerError)
+			return
+		}
+	}
+}
+
+// GetMessagesForSessionHandler retrieves all messages for a specific session.
+//
+// This function handles HTTP GET requests at the /api/v1/session/{sessionId}/messages endpoint.
+// It uses the session ID provided in the URL to fetch all messages associated with that session.
+//
+// The function responds with a JSON array of messages. Each message in the array includes its ID, content, and metadata.
+// If the session ID does not exist, the function responds with a 404 Not Found status code.
+// If there is an error while fetching the messages, the function responds with a 500 Internal Server Error status code.
+//
+// @Summary Retrieves all messages for a specific session
+// @Description get messages by session id
+// @Tags messages
+// @Accept json
+// @Produce json
+// @Param sessionId path string true "Session ID"
+// @Success 200 {array} Message
+// @Failure 404 {object} APIError "Not Found"
+// @Failure 500 {object} APIError "Internal Server Error"
+// @Router /api/v1/session/{sessionId}/messages [get]
+func GetMessagesForSessionHandler(appState *models.AppState) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		sessionID := chi.URLParam(r, "sessionId")
+
+		var limit int
+		var err error
+		if limit, err = handlertools.IntFromQuery[int](r, "limit"); err != nil {
+			limit = DefaultMessageLimit
+		}
+
+		var cursor int
+		if cursor, err = handlertools.IntFromQuery[int](r, "cursor"); err != nil {
+			cursor = 1
+		}
+
+		log.Debugf("GetMessagesForSessionHandler - SessionId %s Limit %d Cursor %d", sessionID, limit, cursor)
+
+		messages, err := appState.MemoryStore.GetMessageList(r.Context(), sessionID, 1, 2)
```

Comment:
Looks like the limit and cursor are hard-coded here?

Reply:
You're absolutely right! Sorry I missed that